### ISSUE TITLE
Free input always allowed on objects

### ISSUE THREAD

**Issue by kennethlarsen** [aupac/ember-aupac-typeahead#30]
When using objects instead of strings as suggestions free input is always allowed and `allowFreeInput` doesn't have any effect.

**Comment by kennethlarsen** [comment]
If #29 was implemented something like this would work:

```
    t.on('focusout', run.bind(this, (/*jqEvent*/) => {
      //the user has now left the control, update display with current binding or reset to blank
      const model = this.get('selection');
      if (this.get('allowFreeInput')) {
        const value = this.get('_typeahead').typeahead('val');
        this.set('selection', value);
        this.sendAction('action', value);
      } else if (model !== null && typeof model === 'object') {
        this.setValue(model);
      } else {
        this.get('_typeahead').typeahead('val', null);
      }
    }));
```

**Comment by jackmatt2** [comment]
Fixed by #31
